crates/data-generator/src/lib.rs

```diff
@@ -0,0 +1,17 @@
+use std::io::Write;
+
+use thiserror::Error;
+
+mod blocks;
+
+#[derive(Debug, Error)]
+pub enum GenerateError {
+    #[error("I/O error: {0}")]
+    Io(#[from] std::io::Error),
+    #[error("error generating blocks: {0}")]
+    Blocks(#[from] blocks::GenerateError),
+}
```

Comment:
As part of the code style unification, don't use `{0}` anywhere across the code base.